templates/commit-msg.branch-name

```diff
@@ -8,12 +8,42 @@ if [ -z "$BRANCHES_TO_SKIP" ]; then
   BRANCHES_TO_SKIP=(master develop test main)
 fi
 
+
 BRANCH_NAME=$(git symbolic-ref --short HEAD)
 
 BRANCH_EXCLUDED=$(printf "%s\n" "${BRANCHES_TO_SKIP[@]}" | grep -c "^$BRANCH_NAME$")
 BRANCH_IN_COMMIT=$(grep -c "\[$BRANCH_NAME\]" $1)
 
-if [ -n "$BRANCH_NAME" ] && ! [[ $BRANCH_EXCLUDED -eq 1 ]] && ! [[ $BRANCH_IN_COMMIT -ge 1 ]]; then 
+# Get the commit template path from git config
+COMMIT_TEMPLATE_PATH=$(git config --get commit.template)
+
+# Check if the commit template path is set
+if [ -n "$COMMIT_TEMPLATE_PATH" ]; then
+  # Read the commit message file and filter out comment lines and empty lines
+  COMMIT_MSG=$(grep -v '^#' "$1" | sed '/^[[:space:]]*$/d')
+
+  # Expand the commit template path
+  COMMIT_TEMPLATE_PATH=$(eval echo "$COMMIT_TEMPLATE_PATH")
+
+  # Read the commit template file
+  if [ -f "$COMMIT_TEMPLATE_PATH" ]; then
+    COMMIT_TEMPLATE=$(cat "$COMMIT_TEMPLATE_PATH")
+  else
+    echo "Commit template file not found at $COMMIT_TEMPLATE_PATH"
+    exit 1
+  fi
+
+  # Filter out comment lines and empty lines from the commit template
+  COMMIT_TEMPLATE=$(echo "$COMMIT_TEMPLATE" | grep -v '^#' | sed '/^[[:space:]]*$/d')
+
+  # Check if the commit message matches the commit template
+  if [ "$COMMIT_MSG" = "$COMMIT_TEMPLATE" ]; then
+    echo "Aborting commit due to default commit message."
+    exit 1
+  fi
+fi
```

Comment:
Another thing githook.py does, which I think is useful, is:
```
if [ -z "$COMMIT_MSG" ]; then
    echo "Aborting commit due to empty commit message."
    exit 1
fi
```
I do this sometimes when I'm creating a commit message and want to give up in the middle.